### PR TITLE
Stats: Add checklist banner component.

### PR DESCRIPTION
**Pull request by taggon**
The component shows the next task in the onboarding checklist.

See p5XAZ9-1wt-p2 for more details on the checklist.

**The banner shows up in only `stage` or `development` environment.**

## How to test

1. Create a new site.
2. Switch to the new site on WordPress.com and select _Stats_ menu on the left sidebar.
3. You should be able to see the banner between the tabs and the chart as follows:
desktop
mobile-banner2
4. When you complete all tasks(or match the `completed` prop with `total`), you will see the completed banner.
completed-banner
5. Each social icon button pops up a window to share your website on social media.
6. If you click on the close button at the top-right of the banner, the banner never shows up again. Alternatively, you can hide the banner by setting `localStorage.setItem( 'onboarding-checklist-banner-closed', true );`

### CODE REVIEW

File: client/my-sites/stats/checklist-banner/index.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		}
+
+		return (
+			<Card className="checklist-banner">
+				<Gauge
+					width={ 152 }
+					height={ 152 }
+					lineWidth={ 18 }
+					percentage={ +percentage }
+					metric={ translate( 'completed' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/checklist-banner/index.jsx]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Show completed' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/checklist-banner-to-stats-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/checklist-banner/index.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+	renderShareButtons() {
+		return <div className="checklist-banner__actions">Share buttons will be here!</div>;
+	}
+
+	renderTaskButton() {
+		return (
+			<div className="checklist-banner__actions">
+				<Button onClick={ this.handleClick } className="checklist-banner__button" primary>
+					{ this.props.translate( 'Do it' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/checklist-banner/index.jsx]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Do it!' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by markryall** [comment]
This looks great - we could merge this to master (once design is reviewed) and deploy if you add a `config.isEnabled` feature toggle.

**Comment by fditrapani** [comment]
Great start @taggon. Other than the `setup` copy it's coming along nicely. 

> config.isEnabled
+1 to that!

**Comment by fditrapani** [comment]
It would be nice to add this component to the devdocs when it's done.

File: client/my-sites/stats/checklist-banner/index.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		return (
+			<Card className="checklist-banner">
+				<div className="checklist-banner__gauge">
+					<span className="checklist-banner__gauge-additional-text">{ translate( 'setup' ) }</span>
+					<Gauge
+						width={ 152 }
+						height={ 152 }
+						lineWidth={ 18 }
+						percentage={ +percentage }
+						metric={ translate( 'completed' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/checklist-banner/index.jsx]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Show completed' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/checklist-banner-to-stats-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/checklist-banner/index.jsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		const percentage = ( completed / Math.max( total, 1 ) * 100 ).toFixed( 1 );
+
+		if ( ! task ) {
+			return siteId && <QuerySiteChecklist siteId={ siteId } />;
+		}
+
+		return (
+			<Card className="checklist-banner">
+				<div className="checklist-banner__gauge">
+					<span className="checklist-banner__gauge-additional-text">{ translate( 'setup' ) }</span>
```
Comment (a8ci18n) [inline client/my-sites/stats/checklist-banner/index.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Setup' )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/checklist-banner-to-stats-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by taggon** [comment]
It's ready for review! @markryall @fditrapani

**Comment by fditrapani** [comment]
The banner looks great @taggon!

**Pull request merged**